lineman/app/services/form_service.coffee

```diff
@@ -0,0 +1,27 @@
+angular.module('loomioApp').factory 'FormService', ($rootScope, FlashService) ->
+  new class FormService
+
+    errorTypes =
+      400: 'badRequest'
+      401: 'unauthorizedRequest'
+      403: 'forbiddenRequest'
+      404: 'resourceNotFound'
+      422: 'unprocessableEntity'
+      500: 'internalServerError'
+
+    submit: (scope, model, options = {}) ->
+      submitFn = options.submitFn or model.save
+      ->
+        scope.isDisabled = true
+        submitFn(model).then ->
```

Comment:
This has a mildly awkward side effect of calling `model.save(model)` by default, which is weird but works fine because save isn't expecting arguments.

Going forward the better solution is to move the `save` function off the model and onto the records interface, which will allow us to just pass the name of a function.

```
FormService.submit $scope, model, 'deactivate'
```
```
submit: (scope, model, action = 'save', options = {}) ->
  ->
    Records[model.constructor.plural][action]().then ->
      # etc
```

This has the added benefit of allowing most forms to operate with a Records dependency, which gets moved to the FormService.